orderer/consensus/etcdraft/chain.go

```diff
@@ -1489,3 +1527,45 @@ func (c *Chain) triggerCatchup(sn *raftpb.Snapshot) {
 	case <-c.doneC:
 	}
 }
+
+// checkForEvictionNCertRotation checks for node eviction and
+// certificate rotation, return true if request includes it
+// otherwise returns false
+func (c *Chain) checkForEvictionNCertRotation(env *common.Envelope) bool {
+	payload, err := protoutil.UnmarshalPayload(env.Payload)
+	if err != nil {
+		c.logger.Warnf("failed to extract payload from config envelope: %s", err)
+		return false
+	}
+
+	configUpdate, err := configtx.UnmarshalConfigUpdateFromPayload(payload)
+	if err != nil {
+		c.logger.Warnf("could not read config update: %s", err)
+		return false
+	}
+
+	configMeta, err := MetadataFromConfigUpdate(configUpdate)
+	if err != nil || configMeta == nil {
+		c.logger.Warnf("could not read config metadata: %s", err)
+		return false
+	}
+
+	membershipUpdates, err := ComputeMembershipChanges(c.opts.BlockMetadata, c.opts.Consenters, configMeta.Consenters)
+	if err != nil {
+		c.logger.Warnf("illegal configuration change detected: %s", err)
+		return false
+	}
+
+	if membershipUpdates.RotatedNode == c.raftID {
+		c.logger.Infof("Detected certificate rotation of our node")
+		return true
+	}
```

Comment:
Is it possible to rotate the cert of a 1-node cluster? what happens then when the node tries to abdicate?

Reply:
Good question. I think it will fail. But in any case, a single node Raft cluster is not something fit for production... I think we can leave this as is.

Reply:
Sounds plausible.